omoo_federate/OMOO.py

```diff
@@ -573,33 +591,17 @@ def run(self):
         self.YL0 = csc_matrix(np.delete(Y, slack_bus, axis=0)[:, slack_bus])
         self.Y = csc_matrix(Y)
         del Y
-        import xarray as xr
-
-        def eqarray_to_xarray(eq: EquipmentNodeArray):
-            return xr.DataArray(
-                eq.values,
-                dims=("eqnode",),
-                coords={
-                    "equipment_ids": ("eqnode", eq.equipment_ids),
-                    "ids": ("eqnode", eq.ids),
-                }
-            )
-
-        def measurement_to_xarray(eq: MeasurementArray):
-            return xr.DataArray(
-                eq.values,
-                coords={
-                    "ids": eq.ids
-                }
-            )
 
-        ratings = eqarray_to_xarray(topology.injections.power_real) + 1j * eqarray_to_xarray(topology.injections.power_imaginary)
+        ratings = eqarray_to_xarray(
+            topology.injections.power_real
+        ) + 1j * eqarray_to_xarray(topology.injections.power_imaginary)
         pv_ratings = ratings[ratings.equipment_ids.str.startswith("PVSystem")]
 
-        previous_power_factor = xr.ones_like(pv_ratings.real)
-        previous_pmpp = xr.ones_like(pv_ratings.real)
-        # while granted_time < h.HELICS_TIME_MAXTIME:
         v = measurement_to_xarray(topology.base_voltage_magnitudes)
+
+        voltages = None
+        power_P = None
+        power_Q = None
         while granted_time < 1000:
```

Comment:
what is the significance of 1000 here? will it work for all usecases?

Reply:
I switched to helics MAXTIME instead.